### PR TITLE
Implemented Collaborators API using the new pattern

### PR DESCRIPTION
**Pull request by dhruvsinghal**
@pengwynn please take a look

### CODE REVIEW

File: octokit/collaborators_test.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package octokit
```
Comment (pengwynn) [inline octokit/collaborators_test.go]: As I mentioned in Slack, I'd probably name this file `repo_collaborators.go`.

File: octokit/repositories_test.go [hunk 2]
```diff
@@ -127,9 +130,11 @@ func TestRepositoresService_CreateFork(t *testing.T) {
 	assert.Equal(t, "This is your first repo", repo.Description)
 	assert.False(t, repo.Private)
 	assert.True(t, repo.Fork)
-	assert.Equal(t, "https://api.github.com/repos/octocat/Hello-World", repo.URL)
+	assert.Equal(t, "https://api.github.com/repos/octocat/Hello-World",
+		repo.URL)
 	assert.Equal(t, "https://github.com/octocat/Hello-World", repo.HTMLURL)
-	assert.Equal(t, "https://github.com/octocat/Hello-World.git", repo.CloneURL)
+	assert.Equal(t, "https://github.com/octocat/Hello-World.git",
```
Comment (pengwynn) [inline octokit/repositories_test.go]: Are these just `go fmt` changes?

**Comment by pengwynn** [comment]
Looking good @dhruvsinghal. Gonna tackle the other `GET` and the mutating methods while you're in here?

File: octokit/repositories_test.go (hunk 2 above)
Comment (dhruvsinghal) [inline octokit/repositories_test.go]: Yup

File: octokit/collaborators_test.go (hunk 1 above)
Comment (dhruvsinghal) [inline octokit/collaborators_test.go]: Done.

**Comment by dhruvsinghal** [comment]
Implemented the remaining GET call. @pengwynn Please take a look.

File: octokit/request.go [hunk 3]
```diff
@@ -35,6 +35,12 @@ func (r *Request) Get(output interface{}) (*Response, error) {
 	return r.createResponse(r.Request.Get(), output)
 }
 
+// Get sends a GET request through the given client and returns the response
+// and any associated errors. It does not fill in a response body.
+func (r *Request) GetBodyless() (*Response, error) {
```
Comment (pengwynn) [inline octokit/request.go]: Not seeing why we need a new method here. Can we not make `Get` handle those types of responses?

**Comment by pengwynn** [comment]
Had one question on this one @dhruvsinghal. Ping me and add the `needs-review` label if you need me to take another look.

File: octokit/request.go (hunk 3 above)
Comment (dhruvsinghal) [inline octokit/request.go]: Done.

**Comment by dhruvsinghal** [comment]
@pengwynn Please take a look.

**Pull request merged**